bundles/org.openhab.binding.wemo/src/main/java/org/openhab/binding/wemo/internal/WemoPowerBank.java

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright (c) 2010-2022 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.wemo.internal;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Deque;
+import java.util.Iterator;
+import java.util.concurrent.ConcurrentLinkedDeque;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.core.library.types.QuantityType;
+
+/**
+ * Class for caching and processing historic values for current power.
+ *
+ * @author Jacob Laursen - Initial contribution
+ */
+@NonNullByDefault
+public class WemoPowerBank {
+
+    private final Deque<CacheItem> slidingCache = new ConcurrentLinkedDeque<CacheItem>();
+
+    @Nullable
+    private QuantityType<?> previousCurrentPower = null;
+    private int slidingSeconds;
+
+    private class CacheItem {
+
+        public CacheItem(double power, Instant start) {
+            this.start = start;
+            this.power = power;
+        }
+
+        public Instant start;
+        public @Nullable Instant end;
+        public double power;
```

Comment:
Members are generally more expected before the constructor.

Reply:
Of course - fixed.